service/controller/clusterapi/v29/resource/ipam/create_test.go

```diff
@@ -0,0 +1,109 @@
+package ipam
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func Test_SubnetAllocator(t *testing.T) {
+	testCases := []struct {
+		name         string
+		object       interface{}
+		errorMatcher func(error) bool
+
+		checker   Checker
+		collector Collector
+		persister Persister
+
+		allocatedSubnetMaskBits int
+		networkRange            net.IPNet
+		privateSubnetMaskBits   int
+		publicSubnetMaskBits    int
+	}{
+		{
+			name:         "case 0 allocate first subnet",
+			object:       &v1alpha1.Cluster{},
+			errorMatcher: nil,
+
+			checker:   NewTestChecker(true),
+			collector: NewTestCollector([]net.IPNet{}),
+			persister: NewTestPersister(mustParseCIDR("10.100.0.0/24")),
```

Comment:
The parameters to the test constructors could take a config object with named properties to make it clearer what these parameters are about. That wasn't too important to me. Let me know when you would like to see this. Then we can just extend the code accordingly.